pkg/experiment/query_backend/block_reader.go

```diff
@@ -132,3 +155,18 @@ func validateRequest(req *queryv1.InvokeRequest) (*request, error) {
 	}
 	return &r, nil
 }
+
+func filterInvalidDatasets(b *metastorev1.BlockMeta, tenantMap map[string]struct{}) ([]*metastorev1.Dataset, error) {
```

Comment:
I don't think `Invalid` is good, maybe this works better (open to other suggestions)

```suggestion
func filterNotOwnedDatasets(b *metastorev1.BlockMeta, tenantMap map[string]struct{}) ([]*metastorev1.Dataset, error) {
```

```suggestion
func filterAlienDatasets(b *metastorev1.BlockMeta, tenantMap map[string]struct{}) ([]*metastorev1.Dataset, error) {
```

```suggestion
func filterOtherTenantsDatasets(b *metastorev1.BlockMeta, tenantMap map[string]struct{}) ([]*metastorev1.Dataset, error) {
```